core/src/main/java/org/apache/brooklyn/core/mgmt/persist/DeserializingClassRenamesProvider.java

```diff
@@ -43,49 +37,30 @@
  * See karaf/init/src/main/resources/OSGI-INF/blueprint/blueprint.xml
  */
 @Beta
-public class DeserializingClassRenamesProvider {
-    private static final Logger LOG = LoggerFactory.getLogger(DeserializingClassRenamesProvider.class);
+public class DeserializingClassRenamesProvider extends DeserializingProvider{
 
-    private static final List<ConfigLoader> loaders = Lists.newCopyOnWriteArrayList();
-    static {
-        loaders.add(new ClasspathConfigLoader());
-    }
-    
-    private static volatile Map<String, String> cache;
+    private static DeserializingClassRenamesProvider instance;
 
-    public static List<ConfigLoader> getLoaders() {
-        return loaders;
+    public static DeserializingClassRenamesProvider getInstance(){
+        if (instance == null) instance = new DeserializingClassRenamesProvider();
```

Comment:
Need to be careful about synchronization for singletons. See Bloch's Effective Java second edition, item 3.

I don't think we need to instantiate it lazily - we can do it as soon as this class is loaded. I'd change it to:
```
private static final DeserializingClassRenamesProvider INSTANCE = new DeserializingClassRenamesProvider();
```